apps/intellij/src/main/kotlin/dev/nx/console/generate/NxGeneratorSearchEvrywhereContributor.kt

```diff
@@ -0,0 +1,121 @@
+package dev.nx.console.generate
+
+import com.intellij.icons.AllIcons
+import com.intellij.ide.actions.searcheverywhere.SearchEverywhereContributor
+import com.intellij.ide.actions.searcheverywhere.SearchEverywhereContributorFactory
+import com.intellij.openapi.actionSystem.ActionPlaces
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.components.service
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.codeStyle.NameUtil
+import com.intellij.util.Processor
+import dev.nx.console.generate.ui.DefaultNxGenerateUiFile
+import dev.nx.console.generate.ui.NxGeneratorListCellRenderer
+import dev.nx.console.models.NxGenerator
+import dev.nx.console.models.NxGeneratorOption
+import dev.nx.console.nxls.server.requests.NxGeneratorOptionsRequestOptions
+import dev.nx.console.services.NxlsService
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import javax.swing.ListCellRenderer
+import kotlinx.coroutines.runBlocking
+
+class NxGeneratorSearchEvrywhereContributor : SearchEverywhereContributorFactory<NxGenerator> {
+    override fun createContributor(
+        initEvent: AnActionEvent
+    ): SearchEverywhereContributor<NxGenerator> = NxGeneratorSEContributor(initEvent)
+}
+
+class NxGeneratorSEContributor(val event: AnActionEvent) :
+    SearchEverywhereContributor<NxGenerator> {
+
+    private val project = event.getRequiredData(CommonDataKeys.PROJECT)
+
+    private val generators = runBlocking { project.service<NxlsService>().generators() }
+
+    private val generatorToOptionsCache: ConcurrentMap<NxGenerator, List<NxGeneratorOption>> =
+        ConcurrentHashMap()
+
+    override fun getSearchProviderId(): String = javaClass.name
+
+    override fun getGroupName(): String = "Nx"
+
+    override fun getSortWeight(): Int = 0
+
+    override fun showInFindResults(): Boolean = true
+
+    override fun getElementsRenderer(): ListCellRenderer<in NxGenerator> =
+        NxGeneratorListCellRenderer(
+            myIcon = AllIcons.Actions.GeneratedFolder,
```

Comment:
I like the `new folder` icon, but I'm not sure if it wouldn't be more consistent to use the nx console icon everywhere for visual clarity. Just a thought, idk what you think @Cammisuli @iguissouma

Reply:
I like the idea of a new icon that is different from the Action, the Action icon will be used for main Actions, not repeated for all generators...

Reply:
should we add that too to the run anything generators?